R/seq_linter.R

```diff
@@ -74,17 +74,32 @@ seq_linter <- function() {
       "seq_len"
     )
 
-    lint_message <- ifelse(
-      grepl("seq", dot_expr1, fixed = TRUE),
-       sprintf(
-        "%s(%s) is likely to be wrong in the empty edge case. Use %s(...) instead.",
-        dot_expr1, dot_expr2, replacement
-      ),
-      sprintf(
-        "%s:%s is likely to be wrong in the empty edge case. Use %s() instead.",
-        dot_expr1, dot_expr2, replacement
+    if (length(replacement) > 0L) {
+      lint_message <- ifelse(replacement == "seq_len",
```

Comment:
i think we can avoid the nested `ifelse` by setting `dot_expr2 = "..."` for the `seq_along()` branch:

```r
dot_expr2[replacement == "seq_along"] <- "..."
lint_message <- ifelse(
  grepl("seq", dot_expr1, fixed = TRUE),
  ...
)
```

Reply:
Tried this, but doesn't seem to work. 🤕

Reply:
Feel free to push into the PR if you can get it to work. Maybe I missed something.

Reply:
Btw, this is the second time we have talked about avoiding nested `ifelse()`.

I am wondering if we should introduce a new linter to cover such cases? 
Providing a concrete alternative in lint message will be difficult, though.

Reply:
> I am wondering if we should introduce a new linter to cover such cases?

There's already `lintr::nested_ifelse_linter()`, it's just not a default linter

> Providing a concrete alternative in lint message will be difficult, though.

There's no requirement to do so. The message can suggest alternatives but doesn't need to be concrete. That's one of the main differences vs. styler, in fact -- styler generally has to be a bit more conservative & shouldn't touch code unless there's a clear, specific improvement, whereas lintr can call out code that can be improved without doing the improvement.

Reply:
oh i see, i had missed that `dot_expr2` was used twice. so introduced `dot_expr3` instead

Reply:
> There's already lintr::nested_ifelse_linter(), it's just not a default linter

Oh, missed that completely! Good to know. 

>  introduced dot_expr3 instead

Thanks for simplifying. Much more readable now!

Reply:
Okay, I went through the docs and the code, but I don't see any explanation for why this is not a default linter.
I think it should be.

E.g., in a PR like the current, we wouldn't even have to rely on its usage being caught by manual code review; the linter GHA itself would notify the contributor about the added lint.